de-webapp/src/main/java/org/iplantc/de/server/HttpSessionConfig.java

```diff
@@ -0,0 +1,61 @@
+package org.iplantc.de.server;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.support.PropertySourcesPlaceholderConfigurer;
+import org.springframework.core.env.Environment;
+import org.springframework.data.redis.connection.jedis.JedisConnectionFactory;
+import org.springframework.session.data.redis.config.annotation.web.http.EnableRedisHttpSession;
+
+/**
+ * Created by sriram on 6/16/17.
+ */
+
+@EnableRedisHttpSession
+@Configuration
+public class HttpSessionConfig {
+
+    @Autowired
+    private Environment environment;
+
+    private final Logger LOG = LoggerFactory.getLogger(HttpSessionConfig.class);
+
+
+    @Bean
+    public static PropertySourcesPlaceholderConfigurer propertySourcesPlaceholderConfigurer() {
+        return new PropertySourcesPlaceholderConfigurer();
+    }
+
+    @Bean
+    JedisConnectionFactory jedisConnectionFactory() {
+        JedisConnectionFactory factory = new JedisConnectionFactory();
+        factory.setHostName(getHostName());
+        factory.setPort(getPort());
+        factory.setDatabase(getDbNumber());
+        factory.setUsePool(true);
+        factory.setPassword(getPassword());
+        return factory;
+    }
+
+    private String getHostName() {
+        return environment.getProperty("org.iplantc.discoveryenvironment.redis.host");
```

Comment:
Right, that's what I was figuring. For consul configs, you'll need to update `de.properties.tmpl` in this repo -- that's the file which configures where values from consul are put into the de.properties used for consul configs. The `j2` file in the ansible repo is what makes it work correctly when generated by ansible, e.g. for config images, but the one in this repo is what controls it for consul configs.

Someday we'll figure out how to not need two files that do the same thing :(